abstractions/dotnet/src/RequestInfo.cs

```diff
@@ -33,7 +33,8 @@ public void SetURI(string currentPath, string pathSegment, bool isRawUrl)
                 if(string.IsNullOrEmpty(currentPath))
                     throw new ArgumentNullException(nameof(currentPath));
                 var parseUri = new Uri(currentPath);
-                foreach(var qsp in parseUri.Query.Split('&').Select(x => x.Split('=')).Where(x => !string.IsNullOrEmpty(x[0]))) {
+                var parseQueryString = parseUri.Query[0] == '?' ? parseUri.Query.Substring(1) : parseUri.Query; //remove leading ? if needed
```

Comment:
Not doing this will result in the first element in the store containing a key starting with the `?` character

Reply:
thanks a lot! you could simply do `parseQueryString.TrimStart('?')` too.

Reply:
Thanks! This is now resolved via https://github.com/microsoft/kiota/pull/557/commits/a7319e0ee5a48d8813c1d26cd906505f1a6547c5